### DISCUSSION TITLE
Border trimming/setting to min_mov

### DISCUSSION THREAD

**Discussion by ethanbb** [flatironinstitute/CaImAn#1362] [Q&A]
I've been thinking about the `motion.border_nan` setting. At first, I was happy to use the default setting of "copy," but I noticed that with either "copy" or "min," I see a band of highly-correlated pixels around the image border (which makes sense), and I sometimes get a large number of elongated ROIs within that band. Maybe not a big deal if they are all rejected, but with my current settings some are accepted. So I wondered whether different border settings could help, specifically leaving them as NaN (though maybe cropping would actually be the best?)

I'm wondering about these lines in the main tutorial notebook:

```python
border_to_0 = 0 if mot_correct.border_nan == 'copy' else mot_correct.border_to_0 # trim border against NaNs
mc_memmapped_fname = cm.save_memmap(mot_correct.mmap_file, 
                                        base_name='memmap_', 
                                        order='C',
                                        border_to_0=border_to_0,  # exclude borders, if that was done
                                        dview=cluster)
```

If I understand correctly, this basically overrides the setting for anything other than 'copy' and forces the border to take the min value over the whole movie. (Which also fixes the behavior of 'min' which otherwise, I think, computes a border value for each chunk individually?) My question is, is there a strong reason for this? For example, mesmerize-core doesn't set `border_to_0` in its equivalent code. But when I did end up with some NaNs in the movie passed to CNMF (due to a bug in my code), I got an error. Should `preprocess.check_nan` be turned off if `motion.border_nan` is True?

Thanks,
Ethan

**Comment by kushalkolar** [comment]
you can just exclude the border, you'll almost never get good ROIs at the border

**Reply by ethanbb** [reply]
That makes sense; so you would typically crop or just exclude borders from CNMF? In any case might be good to add to the tutorial? The border options seem like they should take care of this, but since they can result in high correlation they don't really.
